pkg/model/bootstrapscript.go

```diff
@@ -71,6 +71,17 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup) (*fi.ResourceHo
 			}
 			return ""
 		},
+
+		// Pass in configuration for the cluster proxy
+		"ProxyEnv": func() string {
+			if os.Getenv("CLUSTER_HTTP_PROXY") != "" || os.Getenv("CLUSTER_HTTPS_PROXY") != "" {
+				return fmt.Sprintf("export HTTP_PROXY=%s\nexport HTTPS_PROXY=%s\nexport NO_PROXY=%s\n",
+					os.Getenv("CLUSTER_HTTP_PROXY"),
```

Comment:
Yeah, the environment variables are terrible. I just haven't yet taken the time to familiarize myself with apimachinery.

Looking at it now, I think it should be a new top level object. I don't know whether `proxy` or `proxying` is preferred as the object name, but the fields should just be `httpProxy`, `httpsProxy`, and `noProxy`.

I'm not sure what kind of docs are needed, but I'd be happy to work with you